File: nexus-inventory-backend/src/main/java/com/nexus/inventory/dto/DashboardMetricsDTO.java
package com.nexus.inventory.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DashboardMetricsDTO {
    private long totalProducts;
    private long totalStockUnits;
    private long activeOrders;
    private long lowStockCount;
    private List<ProductDTO> lowStockAlerts;
}

File: nexus-inventory-mobile/app/src/main/java/com/nexus/inventory/data/dto/DashboardMetricsDto.kt
package com.nexus.inventory.data.dto

import com.google.gson.annotations.SerializedName
import com.nexus.inventory.domain.model.DashboardMetrics

data class DashboardMetricsDto(
    @SerializedName("totalProducts") val totalProducts: Long,
    @SerializedName("totalStockUnits") val totalStockUnits: Long,
    @SerializedName("activeOrders") val activeOrders: Long,
    @SerializedName("lowStockCount") val lowStockCount: Long,
    @SerializedName("lowStockAlerts") val lowStockAlerts: List<ProductDto>?
)

fun DashboardMetricsDto.toDomain(): DashboardMetrics = DashboardMetrics(
    totalProducts = totalProducts,
    totalStockUnits = totalStockUnits,
    activeOrders = activeOrders,
    lowStockCount = lowStockCount,
    lowStockAlerts = lowStockAlerts?.map { it.toDomain() } ?: emptyList()
)
